### ISSUE TITLE
Carryalls shouldn't drop harvester next to worms

### ISSUE THREAD

**Issue by abcdefg30** [OpenRA/OpenRA#7310]
Especially as you can't control them and your harvester is instantly worm food.

**Comment by penev92** [comment]
We can make the carryalls query the `WormManager` to ask if there any worms near a location before dropping the harvester there, but what do the carryalls do if there is and they can't drop the harvester?

**Comment by Phrohdoh** [comment]
This should only happen if the worm is visible (or the little electric anim), otherwise this can be misleading and a bit unfair to the hungry worms (and bogus from a logical standpoint).

**Comment by MatthijsBenschop** [comment]
Its the harvester (code) that decides where it will go. So this applies more to harvesters in general, as the same would happen when he's just driving there without the aid of carryall.

**Comment by penev92** [comment]
Yes. And I'm not sure we should be making the harvester code any more complex with worm checks. Also the harvester code is used by all mods. Although I agree with your suggestion, I can't currently think of a decent way to do it.
